deployment/elasticsearch/roundtripper.go

```diff
@@ -0,0 +1,79 @@
+package elasticsearch
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/mattermost/mattermost-load-test-ng/deployment"
+	"github.com/mattermost/mattermost-load-test-ng/deployment/terraform/ssh"
+)
+
+// elasticsearchRoundTripper implements RoundTrip to use it as a Transport in
+// an http client. It signs the requests using AWS Signature Version 4 before
+// letting the request go through its inner transport's RoundTrip.
+type elasticsearchRoundTripper struct {
```

Comment:
Wondering if it's possible to have any tests for this file?

Reply:
Will try to come up with something, at least for the signing and maybe for the correct usage of the Dial function.